### DISCUSSION TITLE
Event 'workflow_run' is only triggered once

### DISCUSSION THREAD

**Discussion by eladchen** [community/community#25484] [Actions]
<p>Hi :wave:t2:.</p>
<p>I’ve been playing around with the <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_run" rel="noopener nofollow ugc">workflow_run</a> event, and came across somewhat of an issue/unexpected behaviour.</p>
<p>Given a workflow like this:</p>
<pre><code class="lang-auto">name: "workflow run webhook"

on:
  workflow_run:
    workflows: ["do something"]
    types: 
      - requested
      - completed

jobs:
  job:
    name: Handle workflow_run webhook
    runs-on: ubuntu-latest
    steps:
      - run: echo "${{ toJson(github.event) }}"
</code></pre>
<p><strong>Expected</strong>:<br>
The above workflow, will only run when another workflow (named “do something”), will be “requested” or “completed”.</p>
<p>Assuming a workflow run named “do something” has being requested, there will be two additional workflow runs triggered, one for the “requested” phase, and another one for the “completed” phase.</p>
<p><strong>Unexpected:</strong><br>
When <strong>re-running</strong> the completed workflow (the one named “do something”) I would expect another two workflow runs to be triggered, one for the “requested” phase, and another for the “completed” phase, and yet, only one workflow triggered, and its the one for the “completed” phase.</p>
<p>So, to summarise, re-running a workflow, I would expect two additional workflow runs, and not just one.</p>
<p>P.S<br>
The above is also relevant when registering for a “workflow_run” webhook.<br>
I’ve also setup repo to demo the above scenario:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/eladchen/workflow-run-webhook-demo/actions" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/3628913/181091346-abc242f2-bf85-4f26-af22-786d56eff247.png" class="thumbnail onebox-avatar" width="200" height="200">

<h3><a href="https://github.com/eladchen/workflow-run-webhook-demo/actions" target="_blank" rel="noopener nofollow ugc">eladchen/workflow-run-webhook-demo</a></h3>

<p>Contribute to eladchen/workflow-run-webhook-demo development by creating an account on GitHub.</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/eladchen">@eladchen</a> ,</p>
<p>Thanks for your feedback.</p>
<p>I can reproduce the same behavior as you reported.<br>
Looks like, ‘<strong>re-run</strong>’ is not considered as an ‘<strong>requested</strong>’. Only triggering a workflow run at the first time can be considered as ‘<strong>requested</strong>’.</p>
<p>If you really need the feature that <strong>workflow_run</strong> event can support the activity type ‘<strong>re-run</strong>’, I recommend that you can directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="noopener nofollow ugc">here</a>.<br>
That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by eladchen** [comment]
<p>Thank you, I’ll do that.</p>

**Comment by eladchen** [comment]
<p>It just stroke me that this is not a feature request, but more of a bug (“workflow._run.completed” does trigger when “re-run”)</p>
<p>How can I report this as a bug…?</p>
